### PR TITLE
RHEL8 updates

### PR DESCRIPTION
**Pull request by Loquacity**
From https://github.com/SUSE/doc-susemanager/pull/885:

- [x] RH8 column
- [x] CentOS 8 column
- [x] supported clients
- [x] Admonition about salt client

From https://bugzilla.suse.com/show_bug.cgi?id=1159023:

- [x] RHEL8 additional packages

### CODE REVIEW

**Comment by Loquacity** [comment]
@juliogonzalez can you please help me with the actual channel names required for RHEL8?

**Comment by Loquacity** [comment]
@mcalmer can you please check that I've re-created the matrices OK?

File: modules/client-configuration/pages/clients-rh.adoc [hunk 1]
```diff
@@ -252,6 +260,13 @@ From the command prompt, add [systemitem]``rhel-x86_64-server-6`` and [systemite
 From the {webui}, add [systemitem]``RHEL7 Base x86_64`` and [systemitem]``SUSE Linux Enterprise Client Tools RES7 x86_64``.
 +
 From the command prompt, add [systemitem]``rhel-x86_64-server-7`` and [systemitem]``res7-suse-manager-tools-x86_64``.
++
+* For {es} 8:
++
+From the {webui}, add [systemitem]``RHEL8 Base x86_64`` and [systemitem]``SUSE Linux Enterprise Client Tools RES8 x86_64``.
+// check what the correct name for the AppStream channel is. LKB 2019-12-13
++
+From the command prompt, add [systemitem]``rhel-x86_64-server-8`` and [systemitem]``res8-suse-manager-tools-x86_64``.
```
Comment (mcalmer) [inline modules/client-configuration/pages/clients-rh.adoc]: Not sure about this.

When I am right, this chapter is for nativ redhat but the var  {es} is used (Expanded support?)
Please check the displayed titles with the reality.  Afaik the tools product for v8 is named different
Please also check differences betwen products and channels. The UI enables products. For v8 these are multiple channels and automatic enables also recommended products.

File: modules/client-configuration/pages/supported-features-centos.adoc [hunk 2]
```diff
@@ -20,52 +20,48 @@ The icons in this table indicate:
...
+| System custom states | ``Salt`` | ``Salt`` | ``Salt``
+| Group custom states | ``Salt`` | ``Salt`` | ``Salt``
+| Organization custom states    | ``Salt`` | ``Salt`` | ``Salt``
+| System set manager (SSM) | icon:check[role="green"] | icon:check[role="green"] | ``Salt``
+| Service pack migration | icon:times[role="danger"] | icon:times[role="danger"] | icon:times[role="danger"]
+| Basic Virtual Guest Management icon:asterisk[role="none"] | ``Traditional`` | icon:check[role="green"] | ``Salt``
+| Advanced Virtual Guest Management icon:asterisk[role="none"] | icon:times[role="danger"] | ``Salt`` | ``Salt``
+| Virtual Guest Installation (Kickstart), as Host OS | ``Traditional`` | ``Traditional`` | icon:times[role="danger"]
+| Virtual Guest Installation (image template), as Host OS | ``Traditional`` | icon:check[role="green"] | ``Salt``
+| System redeployment ({kickstart}) | ``Traditional`` | icon:check[role="green"] | ``Salt``
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features-centos.adoc]: System deployment (kickstart) is missing here. Redhat page has it. Merge error, or did i forget it?

**Comment by juliogonzalez** [comment]
BTW: I suggest you add Ornela and we wait until Monday for merge. In the end she tested this. And maybe whoever added support for RHEL8/RES8.

File: modules/client-configuration/pages/supported-features-centos.adoc (hunk 2 above)
Comment (Loquacity) [reply modules/client-configuration/pages/supported-features-centos.adoc]: Looks like I missed it. Added it in :)

**Comment by Loquacity** [comment]
> BTW: I suggest you add Ornela and we wait until Monday for merge. In the end she tested this. And maybe whoever added support for RHEL8/RES8.

Yeah, I can't @ include her in this repo, it seems. As long as you don't mind waiting for the package, we can hold off until she is back in the office. I will drop her an email so hopefully she sees it when she gets in. Otherwise, would you prefer an admonition or something?

**Comment by juliogonzalez** [comment]
We can wait for Ornela until Monday.

File: modules/client-configuration/pages/clients-sleses.adoc [hunk 3]
```diff
@@ -29,6 +29,13 @@ ifeval::[{uyuni-content} == true]
 endif::[]
 
 
+[IMPORTANT]
+====
+Traditional clients are not available on {rhel}{nbsp}8.
+{rhel}{nbsp}8 clients are only supported as Salt clients.
+====
+
+
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-sleses.adoc]: This is not enough. We need to tell the users how to add Appstream repository and channel. And I am pretty sure the paths for os are now BaseOS. Ornela should now.

File: modules/client-configuration/pages/clients-centos.adoc [hunk 4]
```diff
@@ -20,6 +20,12 @@ ifeval::[{suma-content} == true]
 endif::[]
 
 
+[IMPORTANT]
+====
+Traditional clients are not available on {centos}{nbsp}8.
+{centos}{nbsp}8 clients are only supported as Salt clients.
+====
+
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-centos.adoc]: I am trying to add the channels for CentOS8 at https://github.com/SUSE/spacewalk/pull/10338. Will send the names on Monday, if @paususe decides he wants this into 4.0.4

File: modules/client-configuration/pages/clients-rh.adoc [hunk 5]
```diff
@@ -252,6 +260,13 @@ From the command prompt, add [systemitem]``rhel-x86_64-server-6`` and [systemite
 From the {webui}, add [systemitem]``RHEL7 Base x86_64`` and [systemitem]``SUSE Linux Enterprise Client Tools RES7 x86_64``.
 +
 From the command prompt, add [systemitem]``rhel-x86_64-server-7`` and [systemitem]``res7-suse-manager-tools-x86_64``.
++
+* For {es} 8:
++
+From the {webui}, add [systemitem]``RHEL8 Base x86_64`` and [systemitem]``SUSE Linux Enterprise Client Tools RES8 x86_64``.
+// check what the correct name for the AppStream channel is. LKB 2019-12-13
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-rh.adoc]: Now I know what you mean. The Appstream channel does NOT come from SUSE. It needs to be imported from the DVD, so you need to add the instructions below.

File: modules/client-configuration/pages/clients-centos.adoc (hunk 4 above)
Comment (juliogonzalez) [reply modules/client-configuration/pages/clients-centos.adoc]: Channel for CentoS8:
```
centos8 centos8-appstream centos8-uyuni-client
```
BTW, we didn't mention channels for CentOS6, so maybe we should add the,
```
centos6 centos6-uyuni-client
```
And there's a typo at centos7, the client channel is duplicated.

Comment (juliogonzalez) [reply modules/client-configuration/pages/clients-centos.adoc]: As strange as it sounds, it seems that centos8 doesn't have a separate updates repositories anymore.

File: modules/client-configuration/pages/clients-rh.adoc (hunk 1 above)
Comment (Loquacity) [reply modules/client-configuration/pages/clients-rh.adoc]: we note there that `Your {susemgr} subscription entitles you to the tools channels for {sleses} (also known as {redhat} Expanded Support or RES).`

File: modules/client-configuration/pages/clients-sleses.adoc [hunk 6]
```diff
@@ -81,6 +88,12 @@ From the command prompt, add [systemitem]``rhel-x86_64-server-6`` and [systemite
 From the {webui}, add [systemitem]``RHEL7 Base x86_64`` and [systemitem]``SUSE Linux Enterprise Client Tools RES7 x86_64``.
 +
 From the command prompt, add [systemitem]``rhel-x86_64-server-7`` and [systemitem]``res7-suse-manager-tools-x86_64``.
+* For {es} 8:
++
+From the {webui}, add [systemitem]``RHEL8 Base x86_64`` and [systemitem]``SUSE Linux Enterprise Client Tools RES8 x86_64``.
```
Comment (mcalmer) [inline modules/client-configuration/pages/clients-sleses.adoc]: The Tools channel is named `SUSE Manager Tools for RHEL and ES 8 x86_64` now.

**Comment by jcayouette** [comment]
> > BTW: I suggest you add Ornela and we wait until Monday for merge. In the end she tested this. And maybe whoever added support for RHEL8/RES8.
> 
> Yeah, I can't @ include her in this repo, it seems. As long as you don't mind waiting for the package, we can hold off until she is back in the office. I will drop her an email so hopefully she sees it when she gets in. Otherwise, would you prefer an admonition or something?

I have added ornela

File: modules/client-configuration/pages/clients-sleses.adoc (hunk 6 above)
Comment (jcayouette) [reply modules/client-configuration/pages/clients-sleses.adoc]: Done

File: modules/client-configuration/pages/clients-rh.adoc (hunk 5 above)
Comment (jcayouette) [reply modules/client-configuration/pages/clients-rh.adoc]: Do we still need these instructions?

```
mkdir rhel8
mkdir iso_rhel8
cd iso_rhel8/
wget http://schnell.suse.de/RHEL/RHEL-8/rhel-8.0-x86_64-dvd.iso
cd iso_rhel8/
mount -o loop /iso_rhel8/rhel-8.0-x86_64-dvd.iso /rhel8/
cd /var/lib/rpm
spacewalk-repo-sync -c rhel8-iso-base_os -u file:///rhel8/BaseOS/
cd /rhel8/
# wait for sync to complete...
cd /
cd /var/lib/rpm
spacewalk-repo-sync -c rhel8-iso-app_stream -u file:///rhel8/AppStream/
#wait for sync to complete...
mgr-create-bootstrap-repo --with-custom-channels
1. RHEL8-x86_64
2. SLE-ES8-x86_64
Enter a number of a product label: 1
Creating bootstrap repo for latest Service Pack of RHEL8-x86_64
```

File: modules/client-configuration/pages/clients-rh.adoc [hunk 7]
```diff
@@ -252,6 +267,13 @@ From the command prompt, add [systemitem]``rhel-x86_64-server-6`` and [systemite
 From the {webui}, add [systemitem]``RHEL7 Base x86_64`` and [systemitem]``SUSE Linux Enterprise Client Tools RES7 x86_64``.
 +
 From the command prompt, add [systemitem]``rhel-x86_64-server-7`` and [systemitem]``res7-suse-manager-tools-x86_64``.
++
+* For {es} 8:
++
+From the {webui}, add [systemitem]``RHEL8 Base x86_64`` and [systemitem]``SUSE Linux Enterprise Client Tools RES8 x86_64``.
```
[the remark this answers is not in the document]
Comment (juliogonzalez) [reply modules/client-configuration/pages/clients-rh.adoc]: You should be able to check it at ref40 server.

File: modules/client-configuration/pages/clients-sleses.adoc [hunk 8]
```diff
@@ -29,6 +29,13 @@ ifeval::[{uyuni-content} == true]
 endif::[]
 
 
+[IMPORTANT]
+====
+Traditional clients are not available on {rhel}{nbsp}8.
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-sleses.adoc]: Why RHEL here? This is about SLES ES.

File: modules/client-configuration/pages/clients-sleses.adoc [hunk 9]
```diff
@@ -29,6 +29,13 @@ ifeval::[{uyuni-content} == true]
...
+{rhel}{nbsp}8 clients are only supported as Salt clients.
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-sleses.adoc]: Same.

File: modules/client-configuration/pages/clients-sleses.adoc (hunk 3 above)
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-sleses.adoc]: I don't see where it is explained how to import Appstream from the DVD at this document.

File: modules/client-configuration/pages/clients-rh.adoc (hunk 5 above)
Comment (juliogonzalez) [reply modules/client-configuration/pages/clients-rh.adoc]: We are mixing things. Those instructions are NOT for RHEL, are for SLES ES. The instructions at https://github.com/uyuni-project/uyuni-docs/pull/27/files#diff-f1d37138af48a9d14c0950047b0a724dR164 are still NOT updated.

File: modules/client-configuration/pages/clients-sleses.adoc [hunk 10]
```diff
@@ -81,6 +95,12 @@ From the command prompt, add [systemitem]``rhel-x86_64-server-6`` and [systemite
 From the {webui}, add [systemitem]``RHEL7 Base x86_64`` and [systemitem]``SUSE Linux Enterprise Client Tools RES7 x86_64``.
 +
 From the command prompt, add [systemitem]``rhel-x86_64-server-7`` and [systemitem]``res7-suse-manager-tools-x86_64``.
+* For {es} 8:
++
+From the {webui}, add [systemitem]``RHEL8 Base x86_64`` and [systemitem]``SUSE Linux Enterprise Client Tools RES8 x86_64``.
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-sleses.adoc]: Just in case, go to ref40 and make sure the names are those.

File: modules/client-configuration/pages/clients-sleses.adoc [hunk 11]
```diff
@@ -81,6 +95,12 @@ From the command prompt, add [systemitem]``rhel-x86_64-server-6`` and [systemite
...
+You will also need to add the Appstream channel.
++
+From the command prompt, add [systemitem]``rhel-x86_64-server-8`` and [systemitem]``res8-suse-manager-tools-x86_64``.
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-sleses.adoc]: Same.

File: modules/client-configuration/pages/clients-sleses.adoc [hunk 12]
```diff
@@ -81,6 +95,12 @@ From the command prompt, add [systemitem]``rhel-x86_64-server-6`` and [systemite
 From the {webui}, add [systemitem]``RHEL7 Base x86_64`` and [systemitem]``SUSE Linux Enterprise Client Tools RES7 x86_64``.
 +
 From the command prompt, add [systemitem]``rhel-x86_64-server-7`` and [systemitem]``res7-suse-manager-tools-x86_64``.
+* For {es} 8:
++
+From the {webui}, add [systemitem]``RHEL8 Base x86_64`` and [systemitem]``SUSE Linux Enterprise Client Tools RES8 x86_64``.
+You will also need to add the ``Appstream`` channel.
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-sleses.adoc]: BTW: Why is this here? We don't mention that you need `os` channel for the other versions (and neither to mention the user needs `BaseOS` for SLES ES 8.

I think this can go, I we should focus on `.Procedure: Adding Base Media to Custom Channels`, that is still missing information for SLES ES 8.

File: modules/client-configuration/pages/clients-rh.adoc (hunk 5 above)
Comment (Loquacity) [reply modules/client-configuration/pages/clients-rh.adoc]: We can do that after this release goes out, since we need to update these sections in their entirety anyway. For now, the RH page says that readers need to arrange access to their own RH base media.

File: modules/client-configuration/pages/clients-sleses.adoc (hunk 8 above)
Comment (Loquacity) [reply modules/client-configuration/pages/clients-sleses.adoc]: copypasta, I think.

File: modules/client-configuration/pages/clients-sleses.adoc (hunk 3 above)
Comment (Loquacity) [reply modules/client-configuration/pages/clients-sleses.adoc]: It's not. We explicitly say that the reader needs to organise access to base media. I think we can add this in when we re-organise these pages later on, though.

File: modules/client-configuration/pages/clients-sleses.adoc (hunk 10 above)
Comment (Loquacity) [reply modules/client-configuration/pages/clients-sleses.adoc]: We don't have any RH channels in ref40, I just checked.

File: modules/client-configuration/pages/supported-features.adoc [hunk 13]
```diff
@@ -35,16 +35,15 @@ For details on supported product versions, see https://www.suse.com/lifecycle.
 | {sle} 11 | {x86}, {x86}_64, {ipf}, IBM {ppc}, {ibmz}  | icon:check[role="green"]      | icon:check[role="green"]
 | {sles} for SAP | {x86}_64, {ppc}                       | icon:check[role="green"]      | icon:check[role="green"]
 | {opensuse} Leap 15.1 | {x86}_64                       | icon:times[role="danger"]      | icon:check[role="green"]
-//| {sles}-ES 8 | {x86}_64                                 | icon:check[role="danger"]      | icon:check[role="green"]
+| {sles}-ES 8 | {x86}_64                                 | icon:check[role="danger"]      | icon:check[role="green"]
 | {sles}-ES 7 | {x86}_64                                 | icon:check[role="green"]      | icon:check[role="green"]
 | {sles}-ES 6 | {x86}, {x86}_64                          | icon:check[role="green"]      | icon:check[role="green"]
-//| {rhel} 8 | {x86}_64                                   | icon:times[role="danger"]    | icon:question[role="green"]
+| {rhel} 8 | {x86}_64                                   | icon:times[role="danger"]    | icon:question[role="green"]
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features.adoc]:
```suggestion
| {rhel} 8 | {x86}_64                                   | icon:times[role="danger"]    | icon:check[role="green"]
```

File: modules/client-configuration/pages/supported-features.adoc [hunk 14]
```diff
@@ -35,16 +35,15 @@ For details on supported product versions, see https://www.suse.com/lifecycle.
...
 | {rhel} 7 | {x86}_64                                   | icon:check[role="green"]      | icon:check[role="green"]
 | {rhel} 6 | {x86}, {x86}_64                            | icon:check[role="green"]      | icon:check[role="green"]
-//| {centos} 8 | {x86}_64                                   | icon:times[role="danger"]    | icon:question[role="green"]
+| {centos} 8 | {x86}_64                                   | icon:times[role="danger"]    | icon:question[role="green"]
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features.adoc]:
```suggestion
| {centos} 8 | {x86}_64                                   | icon:times[role="danger"]    | icon:question[role="gray"] / icon:check[role="green"] (with ES)
```

**Comment by Loquacity** [comment]
[uyuni_client_configuration_guide.pdf](https://github.com/uyuni-project/uyuni-docs/files/3977743/uyuni_client_configuration_guide.pdf)

File: modules/client-configuration/pages/supported-features.adoc [hunk 15]
```diff
@@ -35,16 +35,15 @@ For details on supported product versions, see https://www.suse.com/lifecycle.
 | {sle} 11 | {x86}, {x86}_64, {ipf}, IBM {ppc}, {ibmz}  | icon:check[role="green"]      | icon:check[role="green"]
 | {sles} for SAP | {x86}_64, {ppc}                       | icon:check[role="green"]      | icon:check[role="green"]
 | {opensuse} Leap 15.1 | {x86}_64                       | icon:times[role="danger"]      | icon:check[role="green"]
-//| {sles}-ES 8 | {x86}_64                                 | icon:check[role="danger"]      | icon:check[role="green"]
+| {sles}-ES 8 | {x86}_64                                 | icon:check[role="danger"]      | icon:check[role="green"]
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features.adoc]:
```suggestion
| {sles}-ES 8 | {x86}_64                                 | icon:times[role="danger"]      | icon:check[role="green"]
```

File: modules/client-configuration/pages/clients-sleses.adoc [hunk 16]
```diff
@@ -29,6 +29,13 @@ ifeval::[{uyuni-content} == true]
 endif::[]
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-sleses.adoc]: I still don't see where are the instructions to import the repos from DVD below! Only 6 and 7 are mentioned!

File: modules/client-configuration/pages/clients-sleses.adoc (hunk 3 above)
Comment (juliogonzalez) [reply modules/client-configuration/pages/clients-sleses.adoc]: @Loquacity this is already explained step by step for 6 and 7 on this document.

So we need to explain it for 8.

Check the document below, my other comments, or with Joseph.

File: modules/client-configuration/pages/clients-sleses.adoc (hunk 16 above)
Comment (jcayouette) [reply modules/client-configuration/pages/clients-sleses.adoc]: See latest commit. I have added the content to the sleses document.

File: modules/client-configuration/pages/clients-sleses.adoc (hunk 3 above)
Comment (Loquacity) [reply modules/client-configuration/pages/clients-sleses.adoc]: I see it on L100?

Comment (Loquacity) [reply modules/client-configuration/pages/clients-sleses.adoc]: image
(p45 in the PDF I attached)

File: modules/client-configuration/pages/clients-sleses.adoc (hunk 10 above)
Comment (juliogonzalez) [reply modules/client-configuration/pages/clients-sleses.adoc]: We don't have then synced, but you could check at the setup wizard, the product list:

image

This PR is still incorrect.

Comment (juliogonzalez) [reply modules/client-configuration/pages/clients-sleses.adoc]: image

Comment (juliogonzalez) [reply modules/client-configuration/pages/clients-sleses.adoc]: Do not include `(BETA)`, that's going away with this release.

Comment (juliogonzalez) [reply modules/client-configuration/pages/clients-sleses.adoc]: The second screenshot comes from clicking on the "list" icon on the right side of the channel row of the first screenshot.

**Comment by Loquacity** [comment]
:tada:

**Comment by juliogonzalez** [comment]
Not yet :tada: :-)

The PR still has failures (on the comments, and I already told Joseph).

File: modules/client-configuration/pages/clients-sleses.adoc (hunk 3 above)
Comment (juliogonzalez) [reply modules/client-configuration/pages/clients-sleses.adoc]: No, this is NOT what I was talking about. We are talking about **importing the DVD**.

This about adding the channels from SCC for the client tools and RES updates.

Comment (juliogonzalez) [reply modules/client-configuration/pages/clients-sleses.adoc]: But anyway @jcayouette fixed it and added the instructions for DVD import for RHEL/CentOS 8.
